net/kernel_socket.cpp

```diff
@@ -195,26 +202,23 @@ class KernelSocketClient : public SocketClientBase {
 
     ISocketStream* connect(const EndPoint& remote, const EndPoint* local) override {
         sockaddr_storage r(remote);
-        if (likely(!local)) {
+        if (likely(!local || local->is_ipv4() != remote.is_ipv4())) {
```

Comment:
Why we need this check?

Reply:
In case that you bind to an IPv4 address while connecting to an IPv6 address, or vice versa. This is ridiculous.